lib/io/reader.toit

```diff
@@ -309,19 +309,25 @@ abstract class Reader implements old-reader.Reader:
         end := to ? min (start + to) it.size : it.size
         index := it.index-of byte --from=start --to=end
         if index >= 0: return offset + (index - start)
-        if to: to -= it.size - start
+        if to:
+          to -= it.size - start
+          if to <= 0: return -1
         offset += it.size - start
         start = 0
 
+    if to and to <= 0: return -1
+
     while true:
       if not more_:
         if throw-if-absent: throw UNEXPECTED-END-OF-READER
```

Comment:
It isn't very clear what is supposed to happen if we can't find the $byte because $to restricts our search. Should that throw too? The `--throw-if-absent` cases are not very well covered by tests if at all.